### PR TITLE
Minor code cleanup

### PR DESCRIPTION
**Pull request by tangjiangling**
- Remove redundant cast from SmallintType and TinyintType
- Fix grammar and typo in JdbcPageSink

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/type/SmallintType.java [hunk 1]
```diff
@@ -138,7 +138,7 @@ public void appendTo(Block block, int position, BlockBuilder blockBuilder)
     @Override
     public long getLong(Block block, int position)
     {
-        return (long) block.getShort(position, 0);
+        return block.getShort(position, 0);
```
Comment (ebyhr) [inline core/trino-spi/src/main/java/io/trino/spi/type/SmallintType.java:141]: Let's revert this change. Other types also have redundant cast now, e.g, `TinyintType`. Changing only `SmallintType` doesn't make sense to me.

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSink.java [hunk 2]
```diff
@@ -61,10 +61,10 @@ public JdbcPageSink(ConnectorSession session, JdbcOutputTableHandle handle, Jdbc
         }
 
         try {
-            // According to JDBC javaodcs "If a connection is in auto-commit mode, then all its SQL statements will be
+            // According to JDBC javadocs "If a connection is in auto-commit mode, then all its SQL statements will be
```
Comment (ebyhr) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSink.java:64]: Let's fix the commit title as "Fix grammar and typo in JdbcPageSink" if you agree with other comment.

File: core/trino-spi/src/main/java/io/trino/spi/type/SmallintType.java (hunk 1 above)
Comment (tangjiangling) [reply core/trino-spi/src/main/java/io/trino/spi/type/SmallintType.java:141]: Good catch, I checked all the implementations of `AbstractType#getLong` and only `TinyintType` and `SmallintType` have this issue, so I think I should fix all the places instead of revert, do you agree?

Comment (ebyhr) [reply core/trino-spi/src/main/java/io/trino/spi/type/SmallintType.java:141]: Could you separate into two commits? 
* Remove redundant cast from SmallintType and TinyintType
* Fix grammar and typo in JdbcPageSink

**Comment by tangjiangling** [comment]
AC

**Comment by ebyhr** [comment]
CI hit #13556

**Comment by ebyhr** [comment]
Merged, thanks!
